modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/authenticator/JaasAuthenticator.java

```diff
@@ -40,30 +44,48 @@
 public class JaasAuthenticator implements Authenticator {
 
     @Override
-    public void initialize(StartupContext startupContext) throws Exception {
-        UserStoreManager userStoreManager = startupContext.getService(UserStoreManager.class);
-        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
-        BrokerAuthConfiguration brokerAuthConfiguration = configProvider
-                .getConfigurationObject(BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
-        BrokerAuthConfiguration.JaasConfiguration jaasConf = brokerAuthConfiguration.getAuthentication().getJaas();
+    public void initialize(StartupContext startupContext, Map<String, Object> properties) throws Exception {
+
         String jaasConfigPath = System.getProperty(BrokerAuthConstants.SYSTEM_PARAM_JAAS_CONFIG);
+        UserStoreConnector userStoreConnector = new FileBasedUserStoreConnector();
+        userStoreConnector.initialize(startupContext);
+        startupContext.registerService(UserStoreConnector.class, userStoreConnector);
         if (jaasConfigPath == null || jaasConfigPath.trim().isEmpty()) {
-            Configuration jaasConfig = createJaasConfig(jaasConf.getLoginModule(), userStoreManager,
-                                                        jaasConf.getOptions());
-            Configuration.setConfiguration(jaasConfig);
+            Object jaasLoginModule = properties.get(BrokerAuthConstants.CONFIG_PROPERTY_JAAS_LOGIN_MODULE);
+            if (Objects.nonNull(jaasLoginModule)) {
+                // Add user store for default login module
+                if (jaasLoginModule.toString().equals(UserStoreLoginModule.class.getCanonicalName())) {
+                    properties.put(BrokerAuthConstants.PROPERTY_USER_STORE_CONNECTOR,
```

Comment:
I think we need to initialize the userStoreConnector here. that will make sure we initialize it only when we use the UserStoreLoginModule.